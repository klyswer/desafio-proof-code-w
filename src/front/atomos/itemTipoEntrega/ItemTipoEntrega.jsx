import React from "react";
import PropTypes from "prop-types";
import './style.css';

const ItemTipoEntrega = ({tipo,texto}) => {
  return (
    <div className={`item_tipo_entrega ${tipo==='despacho'?'border_blue':'border_orange'}`}>
      <span className={`img_entrega ${tipo==='despacho'?'bg_img_blue':'bg_img_orange'}`}>
        <img
          src={tipo==='despacho'?`${process.env.PUBLIC_URL}/images/despacho-icon.svg`:`${process.env.PUBLIC_URL}/images/delivery-icon.svg`}
          alt="icon-delivery"
        />
      </span>
      <span className={`text_entrega ${tipo==='despacho'?'bg_img_blue':'bg_img_orange'}`}>{texto}</span>
    </div>
  );
};

ItemTipoEntrega.defaultProps = {
  tipo: 'despacho',
  texto: 'despacho',
}

ItemTipoEntrega.propTypes = {
  tipo: PropTypes.string,
  texto: PropTypes.string,
};

export default ItemTipoEntrega;
