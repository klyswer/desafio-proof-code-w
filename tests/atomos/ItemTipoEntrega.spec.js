import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import ItemTipoEntrega from '../../src/front/atomos/itemTipoEntrega/ItemTipoEntrega';

describe('<ItemTipoEntrega />', () => {

  const title = 'retiro';
  const wrapper = shallow( <ItemTipoEntrega texto={title} />);

  test('Primer SnapShot: <ItemSelect />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Caso prueba 1: verificar default props: despacho.', () => {
    const renderItem = wrapper.find('.item_tipo_entrega').prop('className');
    expect(renderItem.includes('border_blue')).toBe(true);
  });

  test('Caso prueba 2: verificar default props: despacho.', () => {
    const renderItem2 = wrapper.find('.item_tipo_entrega span').at(0).prop('className');
    expect(renderItem2.includes('bg_img_blue')).toBeTruthy();
  });

  test('Caso prueba 3: verificar texto:string.', () => {
    const textoEsperado = wrapper.find('.text_entrega').text();
    expect(textoEsperado).toBe(title);
  });
  
  
});