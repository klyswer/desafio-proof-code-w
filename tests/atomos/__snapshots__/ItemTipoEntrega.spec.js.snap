// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemTipoEntrega /> Primer SnapShot: <ItemSelect /> 1`] = `
<div
  className="item_tipo_entrega border_blue"
>
  <span
    className="img_entrega bg_img_blue"
  >
    <img
      alt="icon-delivery"
      src="/images/despacho-icon.svg"
    />
  </span>
  <span
    className="text_entrega bg_img_blue"
  >
    retiro
  </span>
</div>
`;
